src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/WrappingFormattingRule.cs

```diff
@@ -173,25 +175,29 @@ private static (SyntaxToken openBrace, SyntaxToken closeBrace) GetBracePair(Synt
         }
 
         private static void RemoveSuppressOperation(
-            List<SuppressOperation> list,
+            SegmentedList<SuppressOperation> list,
             SyntaxToken startToken,
             SyntaxToken endToken)
         {
             if (startToken.Kind() == SyntaxKind.None || endToken.Kind() == SyntaxKind.None)
-            {
                 return;
-            }
 
             var span = TextSpan.FromBounds(startToken.SpanStart, endToken.Span.End);
-            list.RemoveOrTransformAll(
-                (operation, span) =>
-                {
-                    if (operation.TextSpan.Start >= span.Start && operation.TextSpan.End <= span.End && operation.Option.HasFlag(SuppressOption.NoWrappingIfOnSingleLine))
-                        return null;
+            using var _ = ArrayBuilder<SuppressOperation>.GetInstance(out var copy);
```

Comment:
💡 Another case where rewritten `RemoveOrTransformAll` will be easy to use for in-place transformation

Reply:
yes.  much nicer.  thanks!